pkg/hub/controllers/cluster/reconciler.go

```diff
@@ -155,48 +167,116 @@ func (r *Reconciler) updateClusterInfo(ctx context.Context, cr *hubv1alpha1.Clus
 		if err != nil {
 			return err
 		}
-		if clusterInfo.ClusterProperty.GeoLocation.CloudProvider == GCP || clusterInfo.ClusterProperty.GeoLocation.CloudProvider == AWS || clusterInfo.ClusterProperty.GeoLocation.CloudProvider == AZURE {
-			cr.Spec.ClusterProperty.GeoLocation.CloudRegion = clusterInfo.ClusterProperty.GeoLocation.CloudRegion
-			cr.Spec.ClusterProperty.GeoLocation.CloudProvider = clusterInfo.ClusterProperty.GeoLocation.CloudProvider
-			if err := r.Update(ctx, cr); err != nil {
-				log.Error(err, "Error updating ClusterProperty on hub cluster during retry")
-				return err
+		if clusterInfo.ClusterProperty.GeoLocation.CloudProvider == GCP || clusterInfo.ClusterProperty.GeoLocation.CloudProvider == AWS ||
+			clusterInfo.ClusterProperty.GeoLocation.CloudProvider == AZURE {
+			if clusterInfo.ClusterProperty.GeoLocation.CloudRegion != clusterInfo.ClusterProperty.GeoLocation.CloudRegion ||
```

Comment:
Whats the difference btw the LHS and RHS here?
if clusterInfo.ClusterProperty.GeoLocation.CloudRegion != clusterInfo.ClusterProperty.GeoLocation.CloudRegion
Both are the same.

Reply:
This is a typo, fixing it